src/Workspaces/Core/Portable/RegularExpressions/RegexNodes.cs

```diff
@@ -0,0 +1,1291 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using System.Diagnostics;
+
+namespace Microsoft.CodeAnalysis.RegularExpressions
+{
+    internal sealed class RegexCompilationUnit : RegexNode
```

Comment:
All the nodes we create in the parser.  I considered autogenerating.  But it wasn't that hard to just hand author.  If we get to a point that we want to really flesh these out (for example, adding things like Update methods), then it might make sense to generate.